### PR TITLE
Fail when pulled image with unexpected sha256

### PR DESCRIPTION
**Pull request by alex1545**
To address confusion in #714 when pinning to a manifest list image type.
This forces the user to only pin to a manifest type when using digest.

### CODE REVIEW

File: container/pull.bzl [hunk 1]
```diff
@@ -191,6 +191,14 @@ exports_files(["image.digest", "digest"])
         fail("Failed to read digest: %s" % digest_result.stderr)
     updated_attrs["digest"] = digest_result.stdout
 
+    if repository_ctx.attr.digest and repository_ctx.attr.digest != updated_attrs["digest"]:
+        fail(("SHA256 of the image specified does not match SHA256 of the pulled image. " +
+              "Expected {}, but pulled image with {}. " +
+              "It is possible that you pin to a manifest list which points to another image").format(
```
Comment (nlopezgi) [inline container/pull.bzl]: It is possible that you have a pin to a ...
which points to another image, if so, ... <provide advice as to how to solve>

File: WORKSPACE [hunk 2]
```diff
@@ -82,14 +82,6 @@ container_pull(
     tag = "3.8",
 )
 
-container_pull(
-    name = "k8s_pause_amd64",
-    # this is a manifest list, so the resolved digest should not match this digest
-    digest = "sha256:f78411e19d84a252e53bff71a4407a5686c46983a2c2eeed83929b888179acea",
```
Comment (nlopezgi) [inline WORKSPACE]: lets try to leave this test but remove the digest attr?

Comment (alex1545) [reply WORKSPACE]: Removing the digest attr here will then use the default "latest" tag, which will be hard to test with `file_test`.
And we already have a test that relies on a tag (one target below called `k8s_pause_arm64`).
Let me know if you still want to keep this `container_pull` target without the digest attr and leave it untested.

Comment (nlopezgi) [reply WORKSPACE]: sgtm

**Pull request merged**